File: 21. Merge Two Sorted Lists/MySolution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    if(list1 == null) return list2;
    if(list2 == null) return list1;
    return list1.val < list2.val 
        ? new ListNode(list1.val, mergeTwoLists(list1.next, list2))
        : new ListNode(list2.val, mergeTwoLists(list2.next, list1));
};

File: 21. Merge Two Sorted Lists/MySolution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2) {
        if(list1 == null) return list2;
        if(list2 == null) return list1;
        return list1.val < list2.val 
            ? new ListNode(list1.val, MergeTwoLists(list1.next, list2))
            : new ListNode(list2.val, MergeTwoLists(list1, list2.next));
    }
}